### ISSUE TITLE
Unable to replicate medipipe inference

### ISSUE THREAD

**Issue by asarria13** [google-ai-edge/mediapipe#5709]
### Have I written custom code (as opposed to using a stock example script provided in MediaPipe)

Yes

### OS Platform and Distribution

Ubuntu 22.04.5 LTS

### Python Version

Python 3.11.0rc1

### MediaPipe Model Maker version

0.2.1.4

### Task name (e.g. Image classification, Gesture recognition etc.)

Object detection

### Describe the actual behavior

Unable to replicate mediapipe ObjectDetector.detect() output

### Describe the expected behaviour

Replicate mediapipe ObjectDetector.detect()  output

### Standalone code/steps you may have used to try to get what you need

```shell
I need to run a tflite inference in a device in which I can´t install mediapipe. Using:

detector = ObjectDetector.create_from_options(options)
detection_result = detector.detect(mp_image)

I get an array of 5 detections (the cuantity I declared in options), with its bounding-boxes, score and category, exactly what I want. Unfortunately, as I can´t use mediapipe in the device, I have to run the inference with tensorflow. The output I get in tensorflow is completly different.

The code Im using:

        input_index = interpreter.get_input_details()[0]['index']
        output_details = interpreter.get_output_details()
        
        interpreter.set_tensor(input_index, input_data)
        interpreter.invoke()

        output_data = {}
        for output_detail in output_details:
            tensor_data = interpreter.get_tensor(output_detail['index'])
            output_data[output_detail['name']] = tensor_data

The output I get is the following one: An array of 12276 bounding boxes, and another array with the score of each class (I have two classes) + score of the detection being part of the background I believe.

1. Array:

array([[[ 0.17975327,  0.08251955, -0.6845249 , -0.8700491 ],
        [ 0.14741512,  0.14877419, -0.5271544 , -0.8658894 ],
        [ 0.11626673,  0.23989673, -0.56516916, -1.0368325 ],
        ...,
        [-0.20545518,  0.05216617, -0.36396417,  0.02875823],
        [-0.35858288,  0.02556294, -0.32750204, -0.20667309],
        [-0.43528926, -0.12508349, -0.26987728, -0.06848973]]],
      dtype=float32)

2. Array

array([[[0.00428437, 0.01835944, 0.0032778 ],
        [0.00962438, 0.01591735, 0.01284602],
        [0.00479926, 0.01606906, 0.01060548],
        ...,
        [0.00136925, 0.00643647, 0.007735  ],
        [0.00413275, 0.01283167, 0.00997187],
        [0.00130497, 0.00630908, 0.0065123 ]]], dtype=float32)

Besides, doing the inference with the same model and image in mediapipe and tf scores don´t match, as tf scores are much lower. So I would like to know if there is a way to replicate mediapipes ObjectDetector.detect() to get the same output in tf.
```

**Comment by akashvverma1995** [comment]
Hi @asarria13 

TensorFlow Lite returns the model’s raw output tensors, which is why you see 12,276 anchor predictions with unnormalized scores. MediaPipe ObjectDetector API automatically performs the necessary post-processing steps, such as score normalization and filtering, before returning the final 5 detections such as: 

- Sigmoid Activation: Converts raw classification logits into probability scores.
- Bounding Box Decoding: Translates regression outputs into image coordinates.
- Non-Maximum Suppression (NMS): Filters the thousands of overlapping predictions down to the most confident ones.

To get pure TFLite to behave like MediaPipe on your edge device, you have to recreate that post-processing pipeline yourself. You basically have two options: you can write some custom code in your app to manually parse all those raw arrays after the model runs, or—and this is highly recommended—you can actually bake the Sigmoid and NMS steps directly into your .tflite model graph before you deploy it.


In Python, you can look at [model.py](https://github.com/google-ai-edge/mediapipe/blob/master/mediapipe/model_maker/python/vision/object_detector/model.py)  in the Model Maker export to see how they embed NMS into the model graph.

For the C++ side, check out [tflite_tensors_to_detections_calculator.cc](https://github.com/google-ai-edge/mediapipe/blob/master/mediapipe/calculators/tflite/tflite_tensors_to_detections_calculator.cc) to see how they decode the tensors, apply the activation, and run NMS.
